useful_types/__init__.py

```diff
@@ -216,20 +217,35 @@ def readline(self) -> _T_co: ...
 class SupportsWrite(Protocol[_T_contra]):
     def write(self, __s: _T_contra) -> object: ...
 
+# ====================
+# Buffer protocols
+# ====================
+
+if sys.version_info >= (3, 12):
+    from collections.abc import Buffer as _Buffer
+else:
+    # typing_extensions defines Buffer as an ABC. On older Python versions, we don't have this
+    # special casing: https://github.com/python/cpython/blob/v3.12.0b2/Lib/typing.py#L1764
+    # and so we run into problems when trying to create subprotocols
```

Comment:
You _can_ create a subprotocol from `typing_extensions.Buffer` if you use `typing_extensions.Protocol` ;)

Reply:
https://github.com/python/typing_extensions/blob/cfd49f81b6adf2246b8b3e3f4d3933d19f8e3546/src/typing_extensions.py#L499